internal/mkcw/makefs.go

```diff
@@ -0,0 +1,38 @@
+package mkcw
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+)
+
+// MakeFS formats the imageFile as a filesystem of the specified type,
+// populating it with the contents of the directory at sourcePath.
+// Recognized filesystem types are "ext2", "ext3", "ext4", and "btrfs".
```

Comment:
No support for xfs?

Reply:
There's a comment about `mkfs.xfs` a few lines below.  `mkfs.xfs` doesn't have a "populate with this directory" option.  The closest thing it offers is accepting a protofile, but that doesn't provide the ability to set timestamps on items, or to preserve hard links as hard links.  _Right now_, libkrun also expects us to give it an ext4 filesystem.

Reply:
@slp WDYT?